### PR TITLE
Hosted server

### PR DESCRIPTION
**Pull request by Sorseg**

### CODE REVIEW

File: crates/petri_server/src/plugin.rs [hunk 1]
```diff
@@ -133,8 +133,19 @@ impl Plugin for PetriServerPlugin {
             });
 
             let port = 8989;
+            let hosted_on_fly = std::env::args().any(|a| a == "--flyio");
+
             let current_time = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)?;
-            let public_addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
+            let public_addr = if hosted_on_fly {
```
Comment (demoth) [inline crates/petri_server/src/plugin.rs:139]: pls add a short comment why flyio requires special piece of code inside the app

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# flyctl launch added from .idea/.gitignore
+# Default ignored files
+.idea/shelf
+.idea/workspace.xml
+# Editor-based HTTP Client requests
+.idea/httpRequests
+# Datasource local storage ignored files
+.idea/dataSources
+.idea/dataSources.local.xml
+fly.toml
```
Comment (demoth) [inline .dockerignore:14]:
```suggestion
# Autogenerated
fly.toml
```

File: crates/petri_client/src/plugin.rs [hunk 3]
```diff
@@ -391,6 +397,7 @@ fn send_movement(
 
     let mut direction = Vec2::default();
     // FIXME: figure out the correct directions
```
Comment (demoth) [inline crates/petri_client/src/plugin.rs:399]: done?

File: crates/petri_server/src/plugin.rs (hunk 1 above)
Comment (Sorseg) [reply crates/petri_server/src/plugin.rs]: Done

**Pull request merged**